python/dgl/sparse.py

```diff
@@ -179,15 +190,13 @@ def _gspmm(gidx, op, reduce_op, u, e):
     return v, (arg_u, arg_e)
 
 
-def _gspmm_hetero(g, op, reduce_op, u_and_e_tuple):
+def _gspmm_hetero(g, op, reduce_op, u_len, u_and_e_tuple):
     r""" Generalized Sparse Matrix Multiplication interface.
```

Comment:
This is also getting more complicated so docstrings are needed.

Reply:
Added in backend.py